File: code_samples/scala/testing_fixtures.scala
/**
 * Testing: Fixtures
 * AI/ML Training Sample
 */

class Fixtures {
  private var data: String = ""
  
  def process(input: String): Unit = {
    data = input
  }
  
  def getData: String = data
  
  def validate: Boolean = data.nonEmpty
}

object FixturesExample {
  def main(args: Array[String]): Unit = {
    val instance = new Fixtures()
    instance.process("example")
    println(s"Data: ${instance.getData}")
    println(s"Valid: ${instance.validate}")
  }
}

File: code_samples/go/testing_fixtures.go
package main

import (
    "fmt"
)

// Testing: Fixtures
// AI/ML Training Sample

type Fixtures struct {
    Data string
}

func NewFixtures() *Fixtures {
    return &Fixtures{
        Data: "",
    }
}

func (s *Fixtures) Process(input string) {
    s.Data = input
}

func (s *Fixtures) Validate() bool {
    return len(s.Data) > 0
}

func (s *Fixtures) GetData() string {
    return s.Data
}

func main() {
    instance := NewFixtures()
    instance.Process("example")
    fmt.Printf("Data: %s\n", instance.GetData())
    fmt.Printf("Valid: %v\n", instance.Validate())
}
